aea/test_tools/generic.py

```diff
@@ -119,11 +119,11 @@ def get_nested_ordered_dict_from_keys_and_value(
     root_key = keys[0]
     if (
         isinstance(configuration_obj, SkillConfig)
-        and root_key in SkillConfig.FIELDS_WITH_NESTED_FIELDS
+        and root_key in SkillConfig.FIELDS_ALLOWED_TO_UPDATE
     ):
         root_attr = getattr(configuration_obj, root_key)
         length = len(keys)
-        if length < 3 or keys[2] not in SkillConfig.NESTED_FIELDS_ALLOWED_TO_UPDATE:
+        if length < 3:
```

Comment:
We need a comment in code here explaining the `3` and also make it a constant.

Reply:
> Can we demonstrate in a test that for skills only the desired args are overridable and not the classname?

We already have tests in place

> We need a comment in code here explaining the 3 and also make it a constant. 

I added a comment explaining why we use value `3` but I think adding a constant for this would be a bit too much